auth0/resource_auth0_client.go

```diff
@@ -13,6 +13,14 @@ func newClient() *schema.Resource {
 		Read:   readClient,
 		Update: updateClient,
 		Delete: deleteClient,
+		Importer: &schema.ResourceImporter{
+			State: func(data *schema.ResourceData, i interface{}) ([]*schema.ResourceData, error) {
+				results := make([]*schema.ResourceData, 1, 1)
+				results[0] = data
```

Comment:
These two lines could be simplified into the `[]*ResourceData{data}` literal.

Reply:
Actually, I'm realizing that this approach may not sufficient.

I just dropped my implementation in, but I realized that your `readXXX` functions only populate a subset of the fields into the `ResourceData` object.

Strictly speaking, that's enough to add the client identifiers into the state, but it does mean that the first `terraform plan/apply` will try to apply properties that may already be set.

I guess, if you document that, it's not such a problem.